tools/deploy-with-olm.sh

```diff
@@ -47,24 +58,42 @@ exit_on_err make docker-push IMG=${DEPLOYER_IMAGE}
 # Generate the olm bundle image
 blue "Generate and push the olm bundle image: ${BUNDLE_IMAGE}"
 exit_on_err make bundle IMG=${DEPLOYER_IMAGE}
+exit_on_err sed -i "s|sed_replace_addon_name|${ADDON_NAME}|" ${BUNDLE_FILE}
+exit_on_err sed -i "s|sed_replace_sop_endpoint|${SOP_ENDPOINT}|" ${BUNDLE_FILE}
 exit_on_err make bundle-build BUNDLE_IMG=${BUNDLE_IMAGE}
 exit_on_err make docker-push IMG=${BUNDLE_IMAGE}
 
 # In install olm if needed
 blue "Checking for existing olm"
-${K8S_CLIENT} get namespace olm >/dev/null 2>&1
-if [ $? -eq 0 ]; then
+if ${K8S_CLIENT} get namespace olm >/dev/null 2>&1
+then
     blue "olm found"
+    OLM_NAMESPACE="olm"
+elif ${K8S_CLIENT} get namespace openshift-operator-lifecycle-manager &> /dev/null
+then
+    blue "olm found"
+    OLM_NAMESPACE="openshift-operator-lifecycle-manager"
 else 
     blue "olm not found, installing olm in target cluster"
     exit_on_err ${OP_SDK} olm install     
 fi
 
 # Deploy the bundle
 blue "Deploy/Run the ${BUNDLE_NAME}:${BUNDLE_VERSION} in the cluster (at ${TARGET_NAMESPACE})"
-${K8S_CLIENT} get namespace olm >/dev/null 2>&1
-if [ $? -eq 0 ]; then
-    exit_on_err ${K8S_CLIENT} create namespace ${TARGET_NAMESPACE}
-fi
-exit_on_err ${K8S_CLIENT} apply -f ./tools/ocs-catalogsource.yaml
+
+exit_on_err ${K8S_CLIENT} create namespace ${NAMESPACE} --dry-run=client -o yaml | kubectl apply -f -
+
+exit_on_err sed "s|sed_replace_namespace|${OLM_NAMESPACE}|" ./tools/yamls/ocs-catalogsource.yaml | \
+${K8S_CLIENT} apply -f -
+
+exit_on_err sed "s|sed_replace_addon_name|${ADDON_NAME}|" ${SNITCH_YAML} | \
+sed "s|sed_replace_namespace|${TARGET_NAMESPACE}|" | \
+sed "s|sed_replace_snitch_url|${SNITCH_URL}|" | \
+${K8S_CLIENT} apply -f -
+
+exit_on_err sed -i "s|sed_replace_addon_name|${ADDON_NAME}|" ${PD_YAML} | \
+sed -i "s|sed_replace_namespace|${TARGET_NAMESPACE}|" | \
+sed -i "s|sed_replace_pagerduty_key|${PD_KEY}|" | \
+${K8S_CLIENT} apply -f -
```

Comment:
Why are we trying to configure PD and DMS as part of the deployment?
We can just create the resources and update them, if needed, in another operation after deployment. 
 
 Doing it here just complicated the deployment and add more steps that might fail

Reply:
I added it here because I prefer to have the operator start healthy in an environment where all of its expected prerequisites are set up, which are:
- namespace 
- addon parameter secret
- dms secret
- pd secret

The operator will start in an error loop if PD and DMS are not configured. I don't see why anyone would want that...

If needed I can move these to a install_prerquisites.sh script or something, I just like having to run one script to get what I need done.

Reply:
Ok, I was not clear enough. 
My question is not why are you creating the resources, it meant to be why are you trying to set meaningful values to the fields inside the resources.  

Namespace can be specified as part fo the creation op, while the values of the other fields are irrelevant

Reply:
The script has the _ability_ to set meaningful values to those resources if specified, but the default behavior is a default value. If pd and dms configuration isn't relevant to the run, the user of the script can just omit that configuration; the script will deploy all that is needed.

Reply:
So let's make the default values default, as in empty